run.py

```diff
@@ -56,58 +72,55 @@ def generate_captcha():
-                await asyncio.sleep(3)
-                await client.delete_messages([tmp, message])
-                return
-
-        args = message.content.replace('$' + command + ' ', '', 1)
-        if command in commands.keys():
-            ret = commands[command](message, args)
-            if type(ret) is str:
-                await client.send_message(message.channel, message.author.mention + '\n' + ret)
-
-            elif type(ret) is discord.Embed:
-                await client.send_message(message.channel, message.author.mention, embed=ret)
-
-        elif command in async_commands:
-            await async_commands[command](message, args)
-
-        elif 'karma-store' == message.channel.name:
-            if command in karma_store_cmds:
-                await karma_store_cmds[command](message, args)
-            else:
-                await client.delete_message(message)
+    if all([*map(lambda c: (c in heb), cmd)]):
+        cmd = ''.join([*map(lambda x: (eng[heb.index(x)]), cmd)])
 
+    if cmd in commands.keys():
+        if commands[cmd].channels is None or \
+                any([utils.is_right_channel(message.channel.name, channel) for channel in commands[cmd].channels]):
+            await commands[cmd].function(message, args)
+        else:
+            await client.send_message(message.channel, '**Oops...**  you can\'t use that command in this channel!')
+    else:
+        await client.send_message(message.channel, '**Oops...**  unknown command *{1}{0}* \n'
+                                                   '(use {1}help to see the list of commands)'.format(cmd, CMD_SIGN))
+
+
+@utils.register_command(name='help')
+async def get_help(message, args):
+    """
+    Sends a 'help' message
+    """
+    help_msg = "**__Help:__**\n\n"
+
+    for command_name, command_func, command_channels in utils.register_command.functions_list:
+        if hasattr(command_func, '__doc__') and isinstance(command_func.__doc__, str):
+            doc = command_func.__doc__.split("[END-D]")[0].lstrip().rstrip()
         else:
-            await client.send_message(message.channel,
-                                      message.author.mention + '\n"${}" is not supported!'.format(command))
+            doc = ""
+        help_msg += "**%s%s** - *%s*\n" % (CMD_SIGN, command_name, doc)
+    await client.send_message(message.channel, help_msg)
 
-    elif 'karma-store' == message.channel.name:
-        await client.delete_message(message)
-    if message.author in unverified.keys():
-        if message.content == unverified[message.author]:
-            await client.send_message(message.channel,
-                                      message.author.mention + ' thanks!')
-            unverified.pop(message.author)
 
+@client.event
+async def on_message(message):
+    if message.author == client.user:
+        return
+    elif message.content.startswith(CMD_SIGN):
+        await process_cmd(message)
+    else:
```

Comment:
we stop cursing people for saying "עד מתי"? :(

Reply:
I just was lazy, we can add it again